### PR TITLE
Bug 1247086 - Wrap exception when formatting date/time with missing format string.

### PR DESCRIPTION
**Pull request by jezdez**
This basically works around an issue with Babel's locale data that misses formatting
strings sometimes and raises a KeyError. We fall back to English that such cases.

https://bugzilla.mozilla.org/show_bug.cgi?id=1247086

### CODE REVIEW

**Comment by jezdez** [comment]
@robhudson r?

File: kuma/core/tests/test_helpers.py [hunk 1]
```diff
@@ -167,6 +168,21 @@ def test_unknown_format(self):
...
+        value_english = format_datetime(value_test, locale=u'en_US')
+        self.context['request'].LANGUAGE_CODE = u'fr'
+        mocked_format_datetime.side_effect = [
+            # first call is returning a KeyError as if the format is broken
+            KeyError,
+            # second call returns the English fallback version as expected
+            value_english,
+        ]
+        value_returned = datetimeformat(self.context, value_test,
+                                        format='datetime')
```
Comment (willkg) [inline kuma/core/tests/test_helpers.py]: Why not have it output json and then you get a more "raw" value that you can check more easily?

**Comment by robhudson** [comment]
I think Will's comment is good. r+wc.

File: kuma/core/tests/test_helpers.py (hunk 1 above)
Comment (jezdez) [inline kuma/core/tests/test_helpers.py]: because I don't care

Comment (jezdez) [inline kuma/core/tests/test_helpers.py]: ok, taking care of this. fixing the other tests as well

**Pull request merged**